esmvaltool/preprocessor/_multimodel.py

```diff
@@ -164,133 +177,73 @@ def _get_overlap(cubes):
     """
     all_times = []
     for cube in cubes:
-        # monthly data ONLY
-        # converts all to 365-days calendars
-        time_units = cube.coord('time').units
-        bnd1 = float(cube.coord('time').points[0])
-        bnd2 = float(cube.coord('time').points[-1])
-        if time_units.calendar is not None:
-            if time_units.calendar == '360_day':
-                bnd1 = int(bnd1 / 360.) * 365.
-                bnd2 = (int(bnd2 / 360.) + 1) * 365.
-            else:
-                bnd1 = int(bnd1 / 365.) * 365.
-                bnd2 = (int(bnd2 / 365.) + 1) * 365.
-        else:
-            logger.debug("No calendar type: assuming 365-day")
-            bnd1 = int(bnd1 / 365.) * 365.
-            bnd2 = (int(bnd2 / 365.) + 1) * 365.
+        bnd1 = _datetime_to_int_days(cube)[0]
+        bnd2 = _datetime_to_int_days(cube)[-1]
         all_times.append([bnd1, bnd2])
-    bounds = [range(int(b[0]), int(b[-1]) + 1) for b in all_times]
+    bounds = [range(b[0], b[-1] + 1) for b in all_times]
     time_pts = reduce(np.intersect1d, bounds)
     if len(time_pts) > 1:
         time_bounds_list = [time_pts[0], time_pts[-1]]
         return time_bounds_list
 
 
-def _slice_cube(cube, min_t, max_t):
-    """slice cube on time"""
-    fmt = '%Y-%m-%d-%H'
-    ctr = iris.Constraint(time=lambda x:
-                          min_t <=
-                          datetime.strptime(x.point.strftime(fmt),
-                                            fmt) <= max_t)
-    cube_slice = cube.extract(ctr)
-    return cube_slice
-
-
-def _slice_cube2(cube, t_1, t_2):
+def _slice_cube(cube, t_1, t_2):
     """
     Efficient slicer
 
-    Slice cube on time more memory-efficiently than
-    _slice_cube(); using this function adds virtually
+    Cube indexing has a large memory leak!
+    using this function adds virtually
     no extra memory to the process
     """
     time_pts = [t for t in cube.coord('time').points]
-    converted_t = _time_pts_by_calendar(cube)
+    converted_t = _datetime_to_int_days(cube)
     idxs = sorted([
         time_pts.index(ii) for ii, jj in zip(time_pts, converted_t)
         if t_1 <= jj <= t_2
     ])
-    cube_t_slice = cube.data[idxs[0]:idxs[-1] + 1]
-    return cube_t_slice
+    return [idxs[0], idxs[-1]]
 
 
 def _monthly_t(cubes):
     """Rearrange time points for monthly data"""
     # get original cubes tpoints
     tpts = []
-    # make all 365-day calendards
     for cube in cubes:
-        tpts.append([int(t) for t in _time_pts_by_calendar(cube)])
-
-    # convert to months for MONTHLY data
+        tpts.append(_datetime_to_int_days(cube))
     t_x = list(set().union(*tpts))
-    t_x = list({int((a / 365.) * 12.) for a in t_x})
     t_x.sort()
-    # remake the time axis for roughly the 15th of the month
-    t_x0 = list({t * 365. / 12. + 15. for t in t_x})
+    t_x0 = list({float(t) for t in t_x})
     t_x0.sort()
-
     return t_x, t_x0
```

Comment:
Don't return the same data twice but slightly different from a function. This is confusing and prevents re-usability. Just return the integer time points and convert when needed.

Reply:
good point! done